doc/manual/src/protocols/flake-schemas.md

```diff
@@ -0,0 +1,60 @@
+# Flake Schemas
+
+Flake schemas are a mechanism to allow tools like `nix flake show` and `nix flake check` to enumerate and check the contents of a flake
+in a generic way, without requiring built-in knowledge of specific flake output types like `packages` or `nixosConfigurations`.
+
+A flake can define schemas for its outputs by defining a `schemas` output. `schemas` should be an attribute set with an attribute for
+every output type that you want to be supported. If a flake does not have a `schemas` attribute, Nix uses a built-in set of schemas (namely https://github.com/DeterminateSystems/flake-schemas).
+
+A schema is an attribute set with the following attributes:
+
+* `version`: Should be set to 1.
+* `doc`: A string containing documentation about the flake output type in Markdown format.
+* `allowIFD` (defaults to `true`): Whether the evaluation of the output attributes of this flake can read from derivation outputs.
+* `inventory`: A function that returns the contents of the flake output (described below).
+
+# Inventory
+
+The `inventory` function returns a *node* describing the contents of the flake output. A node is either a *leaf node* or a *non-leaf node*. This allows nested flake output attributes to be described (e.g. `x86_64-linux.hello` inside a `packages` output).
+
+Non-leaf nodes must have the following attribute:
+
+* `children`: An attribute set of nodes. If this attribute is missing, the attribute if a leaf node.
+
+Leaf nodes can have the following attributes:
+
+* `derivation`: The main derivation of this node, if any. It must evaluate for `nix flake check` and `nix flake show` to succeed.
```

Comment:
Some outputs are not technically a `derivation`, but rather a `path`, yet they are paths referring to the output of a derivation. `apps.*.program` for example. How can they be represented in this schema?

Ref: https://github.com/DeterminateSystems/flake-schemas/issues/5